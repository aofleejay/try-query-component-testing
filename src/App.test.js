import React from 'react'
import { MockedProvider } from 'react-apollo/test-utils'
import renderer from 'react-test-renderer'
import waait from 'waait'
import App, { GET_CURRENCIES } from './App'

describe('Test App component', () => {
  it('Should contain loading ui when promise is pending', () => {
    const component = renderer.create(
      <MockedProvider mocks={[]} addTypename={false}>
        <App />
      </MockedProvider>,
    )
  
    expect(component).toMatchSnapshot()
  })

  it('Should contain response when promise has resolved', async () => {
    const mocks = [
      {
        request: {
          query: GET_CURRENCIES,
        },
        result: {
          data: {
            rates: [
              {
                "currency": "AED",
                "rate": "3.67"
              },
            ],
          },
        },
      },
    ]

    const component = renderer.create(
      <MockedProvider mocks={mocks} addTypename={false}>
        <App />
      </MockedProvider>,
    )

    await waait(0)
  
    expect(component).toMatchSnapshot()
  })

  it('Should contain error when promise has rejects', async () => {
    const mocks = [
      {
        request: {
          query: GET_CURRENCIES,
        },
        error: new Error('Noooooooo!!!!')
      },
    ]

    const component = renderer.create(
      <MockedProvider mocks={mocks} addTypename={false}>
        <App />
      </MockedProvider>,
    )

    await waait(0)
  
    expect(component).toMatchSnapshot()
  })
})
