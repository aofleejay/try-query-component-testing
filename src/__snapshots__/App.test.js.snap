// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App component Should contain error when promise has rejects 1`] = `
<p>
  Error :(
</p>
`;

exports[`Test App component Should contain loading ui when promise is pending 1`] = `
<p>
  Loading...
</p>
`;

exports[`Test App component Should contain response when promise has resolved 1`] = `
<div>
  <p>
    AED: 3.67
  </p>
</div>
`;
